### PR TITLE
Add tcollector for map reduce jobs.

### PR DESCRIPTION
**Pull request by mikecdavis**
Adding tcollector leveraging the Application Master [APIs](https://hadoop.apache.org/docs/r2.6.0/hadoop-mapreduce-client/hadoop-mapreduce-client-core/MapredAppMasterRest.html)

Long overdue. This should give us the necessary visibility into our MR jobs.

### CODE REVIEW

File: collectors/30/yarn_mapreduce.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
+import time
+
+from collectors.lib.optimizely_utils import format_tsd_key, get_json
+
+
+# Constants
+TIME = int(time.time())
+HOST_URL = 'http://127.0.0.1:8088'
+HOST_URL = 'http://hbaseprod1-1b-east-0024b866-exclusiverat:8088'
```
Comment (mikeborsuk-optimizely) [inline collectors/30/yarn_mapreduce.py]: Did you mean to leave in the remote url?

File: collectors/30/yarn_mapreduce.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        running_jobs = get_json(job_url)
+
+        for job in running_jobs['jobs']['job']:
+            job_response = get_json(job_url + '/' + job['id'])
+            job_detail = job_response['job']
+
+            job_name = job_detail['name'].lower()
+
+            # Skip oozie jobs
+            if "oozie" in job_name:
```
[the remark this answers is not in the document]
Comment (atuljha-optimizely) [reply collectors/30/yarn_mapreduce.py]: They are the other jobs in the workflow (like taking snapshot, Loading Hfiles etc). They won't have the mapreduce metrics we are looking for.

[the remark this answers is not in the document]
Comment (mikecdavis) [reply collectors/30/yarn_mapreduce.py]: yeah it's just a bit of noise. they are wrapper jobs essentially. You can see them in the RM console and they get in the way :)

File: collectors/30/yarn_mapreduce.py (hunk 1 above)
Comment (mikecdavis) [reply collectors/30/yarn_mapreduce.py]: Nope, fixed-up.

**Pull request merged**